gcf_data_mapper/read.py

```diff
@@ -1,72 +1,165 @@
-import csv
 import json
 import os
 from enum import Enum
-from typing import Any, Optional, Union
+from typing import Optional, Union
 
 import click
+import pandas as pd
 
 
 class AllowedFileExtensions(Enum):
     JSON = "json"
     CSV = "csv"
 
 
-def read_csv(file_path: str) -> list[dict[str, Any]]:
-    """
-    Reads a csv file and returns a list of dictionaries
+def read_csv_pd(
+    file_path: str,
+    header_rows: Optional[Union[int, list[int]]] = 0,
+    chunk_size: int = 10**4,
+) -> pd.DataFrame:
+    """Load the data from the specified CSV file into a Pandas DF.
+
+    :param str file_path: The filepath passed by the user to the
+        tool.
+    :param Optional[Union[int, list[int]]] header_rows: The indexes of
+        any rows in the file that contain headers. Defaults to 0 (which
+        is the first line of the file), but can also be None if no
+        header rows exist or an array of indexes if multiple rows
+        contain headers.
+    :param Optional[int] chunk_size: The number of lines to read into
+        memory in each batch iteratively. Defaults to 10**4.
 
-    :param file_path str: a file path to the csv file
-    :return list: a list of dictionaries, where each line in the csv file is
-    mapped to a dictionary
+    :return Optional[pd.DataFrame]: A Pandas DataFrame containing the
+        CSV data if the file is successfully found and parsed by the
+        Pandas CSV reader. Otherwise this function will return None.
     """
-    with open(file_path, "r") as file:
-        csv_reader = csv.DictReader(file)
-        fieldnames = csv_reader.fieldnames or []
-        data = [{field: line[field] for field in fieldnames} for line in csv_reader]
-        return data
+    # Should the path exist, read the CSV contents into memory iteratively in chunks of
+    # 'chunk_size' (10**4 by default).
+    # This helps prevent out of memory errors where there is insufficient memory to
```

Comment:
An alternative (to using pandas) is to use a generator something like:
```
def _read_csvs(file_path1: str, file_path2: str) -> Generator:
    with open(file_path1) as f1, open(file_path2) as f2:
        reader1 = csv.DictReader(f1)
        reader2 = csv.DictReader(f2)
        for line1, line2 in zip(reader1, reader2):
            yield line1, line2
```
Not that one is more correct over the other, but this way does reduce dependencies :smile:

Reply:
Yeah, appreciate this :) I think pandas, despite introducing more dependencies, really does speed up the reading/retrieval of data (especially with larger datasets) and makes it a lot simpler to merge fields from files of different types :)